### PR TITLE
Add Vip addresses & InstanceEndpoints to VMs

### PR DESCRIPTION
**Pull request by keithchambers**
Credit to @frodenas for inspiration.

CLA has been signed.

### CODE REVIEW

File: clients/vmClient/entities.go [hunk 1]
```diff
@@ -171,6 +186,14 @@ type RoleSize struct {
 	VirtualMachineResourceDiskSizeInMb int
 }
 
+type VirtualIPs struct {
+	VirtualIP []*VirtualIP
```
Comment (ahmetalpbalkan) [inline clients/vmClient/entities.go]: why `*`?

File: clients/vmClient/entities.go [hunk 2]
```diff
@@ -34,11 +35,25 @@ type RoleInstanceList struct {
...
+	InstanceName      string
+	InstanceStatus    string
+	InstanceSize      string
+	PowerState        string
+	IpAddress         string
+	InstanceEndpoints InstanceEndpoints `xml:",omitempty"`
+}
+
+type InstanceEndpoints struct {
+	InstanceEndpoint []*InstanceEndpoint
```
Comment (ahmetalpbalkan) [inline clients/vmClient/entities.go]: why `*`?

Comment (keithchambers) [inline clients/vmClient/entities.go]: Thanks ... removed.

File: clients/vmClient/entities.go (hunk 1 above)
Comment (keithchambers) [inline clients/vmClient/entities.go]: Thanks ... removed.

**Comment by ahmetalpbalkan** [comment]
@keithchambers please squash the additional fixes into one commit and `git push -f` to your branch. 

cc: @hopetobelievein @paulmey @jeffmendoza for CR

**Comment by keithchambers** [comment]
Thank you Ahmet.

I have now squashed the commits and pushed to my branch.

**Comment by jeffmendoza** [comment]
Looks good

**Comment by ahmetalpbalkan** [comment]
Lgtm as well

**Pull request merged**